ui/src/adapters/parsers/forms.ts

```diff
@@ -110,17 +113,28 @@ export interface CredentialFormInput {
   issueCredential: IssueCredentialFormData;
 }
 
-export const credentialFormParser = getStrictParser<CredentialFormInput, IssueCredentialForm>()(
+export const credentialFormParser = getStrictParser<
+  CredentialFormInput,
+  CredentialLinkIssuance | CredentialDirectIssuance
+>()(
   z
     .object({
       issuanceMethod: issuanceMethodFormDataParser,
       issueCredential: issueCredentialFormDataParser,
     })
     .transform(({ issuanceMethod, issueCredential }, context) => {
-      if (issuanceMethod.type === "credentialLink") {
-        const { linkExpirationDate, linkExpirationTime, linkMaximumIssuance, type } =
-          issuanceMethod;
-        const { credentialSubject, expirationDate } = issueCredential;
+      const { credentialSubject, expirationDate, proofTypes } = issueCredential;
+      const { type } = issuanceMethod;
+
+      const baseIssuance = {
+        credentialSubject,
+        expiration: expirationDate ? expirationDate.toDate() : undefined,
+        mtProof: proofTypes.includes("MTP"),
```

Comment:
Can we type this somehow? I am frustrated that there wouldn't be an error if we put `proofTypes.includes("foo")`.

Reply:
This is [what I meant here](https://github.com/0xPolygonID/sh-id-platform/pull/330#discussion_r1176368916). Let's try to improve it after I merge my branch. Leave the comment open.

Reply:
You have an error if you put something different than `MTP`  or `SIG`

Reply:
Yep, you get an error, the problem is having to do the includes, but anyway, this is not urgent.

Reply:
I opened [this PR](https://github.com/0xPolygonID/sh-id-platform/pull/348/files) playing with a possible refactor of the modeling and parsing of the proof types.

Like this, our domain only manages the `type ProofType = "MTP" | "SIG" | "MTP & SIG"` and the parsers take care of transforming the input from the API and the form to this domain type, so the views no longer need to do the "includes" (only the parser do) and have last resort fallbacks (see the switch/case in `CredentialLinkQR.tsx`).

Also, we could simplify the parsing of the form if instead of checkboxes we use radio buttons. This is, IMO, also a better UX because we get rid of that "At least one proof type is required" validation since the field is required and the user is forced to select an option.

Let me know wat you think.

Reply:
[I've asked Liudas about the radio buttons](https://0xpolygon.slack.com/archives/C03HUC0Q7RU/p1682491901800369) and he revealed a fundamental fact I was not aware of:

> In the future there would be more of these proof types, making it even more option if we were to go with radios

This makes the refactor I proposed a bad option for obvious reasons, so I'm closing it.

Reply:
Never mind, it's because my plugin wasn't working properly. It's showing up as expected now, sorry for the confusion.

![Screenshot from 2023-04-26 09-43-36](https://user-images.githubusercontent.com/2691584/234505449-0a88ba7a-5da9-4975-9b73-9fc965ffa807.png)